akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -162,17 +157,39 @@ object Behavior {
    * and then the resulting behavior is returned.
    */
   def start[T](behavior: Behavior[T], ctx: TypedActorContext[T]): Behavior[T] = {
-    // note that this can't be @tailrec, but normal stack of interceptors and similar shouldn't be
-    // that deep, and if they are it's most likely a bug which will be seen as StackOverflowError
-    behavior match {
-      case innerDeferred: DeferredBehavior[T]          => start(innerDeferred(ctx), ctx)
-      case wrapped: InterceptorImpl[T, Any] @unchecked =>
-        // make sure that a deferred behavior wrapped inside some other behavior is also started
-        val startedInner = start(wrapped.nestedBehavior, ctx.asInstanceOf[TypedActorContext[Any]])
-        if (startedInner eq wrapped.nestedBehavior) wrapped
-        else wrapped.replaceNested(startedInner)
-      case _ => behavior
+    @tailrec
+    def internalStart(
+        behavior: Behavior[T],
+        ctx: TypedActorContext[T],
+        interceptorsStack: List[InterceptorImpl[T, Any]] = Nil,
+        maxDepth: Int = 0): Behavior[T] = {
+      // normal stack of interceptors and similar shouldn't be
+      // that deep, and if they are it's most likely a bug which will be seen as StackOverflowError
+      if (maxDepth > 8192) {
```

Comment:
This number can be adjusted and it's more testable.

Reply:
But how is 8192 chosen? And how would that be adjusted? 

I don't think we should have a fixed magical number. We never know in which JVM this will run. And having it adjustable by the user seems to me a leaked abstraction.

Reply:
ah ok, I should have search before commenting. 8192 is the deault StackOverflow size.

Reply:
@renatocaval For myself,I want to remove the limit of depth too, but as a PR, I think it would be great to simulate it's previous behavior.
I just pick up 8192 randomly because we do not know how many frames can be there before it throws a `StackOverflowError`.
>    // note that this can't be @tailrec, but normal stack of interceptors and similar shouldn't be
    // that deep, and if they are it's most likely a bug which will be seen as StackOverflowError

Reply:
I bring this up for discussion, thanks for your inputs.

Reply:
I have updated this and removed the `maxDepth` as it can only be a bug when OOM.